framework/src/arastorage/aql_exec.c

```diff
@@ -170,7 +170,7 @@ db_result_t db_exec(char *format)
 	optype = AQL_GET_EXEC_TYPE(AQL_GET_TYPE(&adt));
 	if (optype != AQL_TYPE_CREATE_RELATION) {
 		rel = aql_get_relation(&adt);
-		if (rel == NULL) {
+		if (rel == NULL && optype != AQL_TYPE_REMOVE_RELATION) {
```

Comment:
if (optype != AQL_TYPE_CREATE_RELATION && optype != AQL_TYPE_REMOVE_RELATION) {
    rel = aql_get_relation();
   .
   .
}

Reply:
When optype is AQL_TYPE_REMOVE_RELATION, aql_get_relation is needed to called for a case of removing existent relation before calling relation_remove